platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/WritableRosterStore.java

```diff
@@ -0,0 +1,181 @@
+        requireNonNull(candidateRoster);
+        RosterValidator.validate(candidateRoster);
+
+        final Bytes incomingCandidateRosterHash =
+                RosterUtils.hash(candidateRoster).getBytes();
+
+        // update the roster state/map
+        final RosterState previousRosterState = rosterStateOrThrow();
+        final Bytes previousCandidateRosterHash = previousRosterState.candidateRosterHash();
+        final Builder newRosterState =
+                previousRosterState.copyBuilder().candidateRosterHash(incomingCandidateRosterHash);
+        removeRoster(previousCandidateRosterHash);
+        storeRoster(candidateRoster, incomingCandidateRosterHash, newRosterState);
+    }
+
+    /**
+     * Sets the Active roster.
+     * This will be called to store a new Active Roster in the state.
+     * The roster must be valid according to rules codified in {@link com.swirlds.platform.roster.RosterValidator}.
+     *
+     * @param roster an active roster to set
+     * @param round the round number in which the roster became active.
+     *              It must be a positive number greater than the round number of the current active roster.
+     */
+    public void putActiveRoster(@NonNull final Roster roster, final long round) {
+        requireNonNull(roster);
+        RosterValidator.validate(roster);
+
+        // update the roster state
+        final RosterState previousRosterState = rosterStateOrThrow();
+        final List<RoundRosterPair> roundRosterPairs = new LinkedList<>(previousRosterState.roundRosterPairs());
+        if (!roundRosterPairs.isEmpty()) {
+            final RoundRosterPair activeRosterPair = roundRosterPairs.getFirst();
+            if (round < 0 || round <= activeRosterPair.roundNumber()) {
+                throw new IllegalArgumentException(
+                        "incoming round number must be greater than the round number of the current active roster.");
+            }
+        }
+        final Bytes activeRosterHash = RosterUtils.hash(roster).getBytes();
+        roundRosterPairs.addFirst(new RoundRosterPair(round, activeRosterHash));
+
+        if (roundRosterPairs.size() > MAXIMUM_ROSTER_HISTORY_SIZE) {
+            final RoundRosterPair lastRemovedRoster = roundRosterPairs.removeLast();
+            removeRoster(lastRemovedRoster.activeRosterHash());
+
+            // At this phase of the implementation, the roster state has a fixed size limit for active rosters.
+            // Future implementations (e.g. DAB) can modify this.
+            if (roundRosterPairs.size() > MAXIMUM_ROSTER_HISTORY_SIZE) {
+                // additional safety check to ensure that the roster state does not contain more than set limit.
+                throw new IllegalStateException(
+                        "Active rosters in the Roster state cannot be more than  " + MAXIMUM_ROSTER_HISTORY_SIZE);
+            }
+        }
+
+        final Builder newRosterState = previousRosterState
```

Comment:
Minor: this object isn't a state as its name currently suggests. It's a builder. I suggest to rename the variable accordingly for clarity.

Reply:
Renamed variable, thanks.